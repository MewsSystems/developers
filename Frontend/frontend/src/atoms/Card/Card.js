import styled from 'styled-components';


const Card = styled.div`
  background: ${(p) => p.theme.white};
  padding: ${(p) => p.theme.common.paddingSM};
  margin-bottom: ${(p) => p.theme.common.paddingMD};
  box-shadow: 6px 3px 12px 0 rgba(0, 0, 0, 0.05);
`;


export default Card;
