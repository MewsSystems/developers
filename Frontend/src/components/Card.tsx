import React from 'react'
import styled from 'styled-components'
import truncate from 'truncate'
import { COLORS } from 'constants/colors'
import { BORDER_RADIUS, BOX_SHADOW } from 'constants/index'

export const StyledCard = styled.div`
  display: flex;
  flex-direction: column;
  flex-grow: 0;
  justify-content: flex-end;
  height: 18rem;
  padding: 1rem;
  background-repeat: no-repeat;
  background-size: cover, cover;
  border: 1px solid ${COLORS.GRAY};
  border-radius: ${BORDER_RADIUS.MEDIUM};
  box-shadow: ${BOX_SHADOW.MEDIUM};
  cursor: pointer;
  opacity: 0.85;
  transition: opacity ease-in 0.3s;

  :hover {
    opacity: 1;
    transition: opacity ease-in 0.3s;
  }
`

const Title = styled.h2`
  color: ${COLORS.WHITE};
  font-size: 1.2rem;
  text-shadow: 1px 1px 0 ${COLORS.BLACK};
`

const TitleContainer = styled.div`
  display: flex;
  flex-direction: row;
  margin: 0 0 0.3rem;

  ${Title} {
    margin-right: 0.8rem;
  }
`

const Overview = styled.p`
  color: ${COLORS.WHITE};
  font-size: 0.8rem;
  text-shadow: 1px 1px 0 ${COLORS.BLACK};
`

const Language = styled.span`
  align-self: center;
  padding: 0.1rem 0.3rem 0.15rem;
  color: ${COLORS.WHITE};
  font-weight: bold;
  font-size: 0.8rem;
  text-transform: uppercase;
  background: ${COLORS.DARK_GRAY};
  border-radius: ${BORDER_RADIUS.MEDIUM};
`

export interface CardProps {
  id: number
  background: string
  title: string
  overview: string
  overviewMaxLength?: number
  language: string
  onClick?: (id: number) => void
}

export const Card: React.FC<CardProps> = ({
  id,
  background,
  title,
  overview,
  overviewMaxLength = 120,
  language,
  onClick,
}) => {
  const backgroundImage = `linear-gradient(
      180deg,
      rgba(0, 0, 0, 0) 0%,
      rgba(0, 0, 0, 0.2) 55%,
      rgba(0, 0, 0, 0.5) 100%
    ),
    url(${background})`

  return (
    <StyledCard
      style={{ backgroundImage }}
      onClick={() => onClick && onClick(id)}
    >
      <TitleContainer>
        <Title>{title}</Title>
        <Language>{language}</Language>
      </TitleContainer>

      <Overview>{truncate(overview, overviewMaxLength)}</Overview>
    </StyledCard>
  )
}
