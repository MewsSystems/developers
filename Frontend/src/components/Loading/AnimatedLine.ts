import styled from 'styled-components'

export interface AnimatedLineProps {
  width: string
  height: string
}

export const AnimatedLine = styled.div<AnimatedLineProps>`
  width: ${({ width }) => width};
  height: ${({ height }) => height};
  background: linear-gradient(
    90deg,
    rgba(195, 195, 195, 1) 0%,
    rgba(246, 246, 246, 1) 100%
  );
  animation: AnimationName 5s ease infinite;

  @keyframes AnimationName {
    0% {
      background-position: 0% 50%;
    }
    50% {
      background-position: 100% 50%;
    }
    100% {
      background-position: 0% 50%;
    }
  }
`
