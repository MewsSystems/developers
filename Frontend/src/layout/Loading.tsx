import * as React from 'react'
import { keyframes } from '@emotion/core'
import { transparentize } from 'polished'
import styled from '../utils/styled'

export const LoadingSpinner: React.SFC = () => (
  <Spinner>
    <div className="cube1" />
    <div className="cube2" />
  </Spinner>
)

const CubeMove = keyframes`
  25% {
    transform: translateX(42px) rotate(-90deg) scale(0.5);
    -webkit-transform: translateX(42px) rotate(-90deg) scale(0.5);
  } 50% {
    transform: translateX(42px) translateY(42px) rotate(-179deg);
    -webkit-transform: translateX(42px) translateY(42px) rotate(-179deg);
  } 50.1% {
    transform: translateX(42px) translateY(42px) rotate(-180deg);
    -webkit-transform: translateX(42px) translateY(42px) rotate(-180deg);
  } 75% {
    transform: translateX(0px) translateY(42px) rotate(-270deg) scale(0.5);
    -webkit-transform: translateX(0px) translateY(42px) rotate(-270deg) scale(0.5);
  } 100% {
    transform: rotate(-360deg);
    -webkit-transform: rotate(-360deg);
  }
`

const Spinner = styled('div')`
  margin: 100px auto;
  width: 40px;
  height: 40px;
  position: relative;

  & .cube1,
  & .cube2 {
    background-color: ${props => props.theme.colors.brand};
    width: 15px;
    height: 15px;
    position: absolute;
    top: 0;
    left: 0;

    -webkit-animation: ${CubeMove} 1.8s infinite ease-in-out;
    animation: ${CubeMove} 1.8s infinite ease-in-out;
  }

  & .cube2 {
    -webkit-animation-delay: -0.9s;
    animation-delay: -0.9s;
  }
`

export const LoadingOverlay = styled('div')`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 10;
  background: ${props => transparentize(0.25, props.theme.colors.background)};
`

export const LoadingOverlayInner = styled('div')`
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 200px;
`

type LoadingProps = {
  loading: boolean
}

export const Loading = ({ loading }: LoadingProps) => {
  if (!loading) return null

  return (
    <LoadingOverlay>
      <LoadingOverlayInner>
        <LoadingSpinner />
      </LoadingOverlayInner>
    </LoadingOverlay>
  )
}
