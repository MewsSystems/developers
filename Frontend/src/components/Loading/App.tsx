import React from 'react'
import styled from 'styled-components'
import { COLORS } from 'constants/colors'
import { Content } from 'components/Content'
import { AnimatedLine } from './AnimatedLine'
import { Spacing } from 'components/Spacing'

const Header = styled.div`
  height: 10.6rem;
  background: ${COLORS.GRAY};
`

export const App: React.FC = () => (
  <>
    <Header />
    <Content>
      <Spacing outer="2rem 0">
        <AnimatedLine width="20rem" height="2rem" />
      </Spacing>
    </Content>
  </>
)
