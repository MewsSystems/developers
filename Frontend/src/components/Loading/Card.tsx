import React from 'react'
import { AnimatedLine } from './AnimatedLine'
import { StyledCard } from 'components/Card'
import { Spacing } from 'components/Spacing'

export const Card: React.FC = () => (
  <StyledCard>
    <Spacing outer="0 0 1rem">
      <AnimatedLine width="100%" height="2rem" />
    </Spacing>
    <AnimatedLine width="100%" height="1rem" />
  </StyledCard>
)
