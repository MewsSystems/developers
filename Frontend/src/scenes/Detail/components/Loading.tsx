import React from 'react'
import { StyledHeroBanner } from 'components/HeroBanner'
import { Content } from 'components/Content'
import { AnimatedLine } from 'components/Loading/AnimatedLine'

export const Loading: React.FC = () => (
  <>
    <StyledHeroBanner />
    <Content>
      <AnimatedLine height="2rem" width="80%" />
    </Content>
  </>
)
