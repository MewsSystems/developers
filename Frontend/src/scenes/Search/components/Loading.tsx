import React from 'react'
import { Content } from 'components/Content'
import { Spacing } from 'components/Spacing'
import styled from 'styled-components'
import { Card } from 'components/Loading/Card'
import { AnimatedLine } from 'components/Loading/AnimatedLine'

const Grid = styled.div`
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: 1fr 1fr 1fr;
`

export const Loading: React.FC = () => (
  <Content>
    <Spacing outer="2rem 0">
      <AnimatedLine width="16rem" height="2rem" />
    </Spacing>
    <Spacing outer="2rem 0">
      <AnimatedLine width="12rem" height="1rem" />
    </Spacing>
    <Grid>
      <Card />
      <Card />
      <Card />
    </Grid>
  </Content>
)
