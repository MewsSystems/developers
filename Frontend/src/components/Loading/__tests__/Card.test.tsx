import { Card } from 'components/Loading/Card'
import { shallow } from 'enzyme'
import React from 'react'

describe('Test Loading Card', () => {
  it('Tests snapshot', () => {
    expect(shallow(<Card />)).toMatchSnapshot()
  })
})
