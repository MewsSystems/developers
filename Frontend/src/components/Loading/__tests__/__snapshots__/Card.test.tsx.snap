// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Loading Card Tests snapshot 1`] = `
<styled.div>
  <Spacing
    outer="0 0 1rem"
  >
    <styled.div
      height="2rem"
      width="100%"
    />
  </Spacing>
  <styled.div
    height="1rem"
    width="100%"
  />
</styled.div>
`;
